### PR TITLE
[arrow] Update to 8.0.0

### PR DESCRIPTION
**Pull request by ianmcook**
Updates the arrow port to version 8.0.0

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@ianmcook ,Thanks for posting this issue, looks ci tested failed on x64-windos triplet, this is error log:
Please take a look:
````
C:\PROGRA~1\MICROS~1\2022\ENTERP~1\VC\Tools\MSVC\1431~1.311\bin\Hostx64\x64\cl.exe   /TP -DARROW_EXPORTING -DARROW_HAVE_RUNTIME_AVX2 -DARROW_HAVE_RUNTIME_AVX512 -DARROW_HAVE_RUNTIME_BMI2 -DARROW_HAVE_RUNTIME_SSE4_2 -DARROW_HAVE_SSE4_2 -DARROW_WITH_BROTLI -DARROW_WITH_BZ2 -DARROW_WITH_LZ4 -DARROW_WITH_RE2 -DARROW_WITH_SNAPPY -DARROW_WITH_TIMING_TESTS -DARROW_WITH_UTF8PROC -DARROW_WITH_ZLIB -DARROW_WITH_ZSTD -DBOOST_ALL_DYN_LINK -DBOOST_ALL_NO_LIB -DBROTLI_SHARED_COMPILATION -DURI_STATIC_BUILD -D_CRT_SECURE_NO_WARNINGS -D_ENABLE_EXTENDED_ALIGNED_STORAGE -Darrow_shared_EXPORTS -ID:\buildtrees\arrow\x64-windows-dbg\src -ID:\buildtrees\arrow\src\rrow-8.0.0-0091dfae41.clean\cpp\src -ID:\buildtrees\arrow\src\rrow-8.0.0-0091dfae41.clean\cpp\src\generated -external:ID:\buildtrees\arrow\src\rrow-8.0.0-0091dfae41.clean\cpp\thirdparty\flatbuffers\include -external:ID:\installed\x64-windows\include -external:ID:\installed\x64-windows\share\thrift\..\..\include\thrift -external:ID:\installed\x64-windows\share\rapidjson\..\..\include -external:ID:\buildtrees\arrow\x64-windows-dbg\xsimd_ep\src\xsimd_ep-install\include -external:ID:\buildtrees\arrow\src\rrow-8.0.0-0091dfae41.clean\cpp\thirdparty\hadoop\include -external:W0 /nologo /DWIN32 /D_WINDOWS  /utf-8 /GR /EHsc /MP  /D_SILENCE_TR1_NAMESPACE_DEPRECATION_WARNING   /EHsc /wd5105 /bigobj /utf-8 /W3 /wd4800 /wd4996 /wd4065   /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /Fosrc\arrow\CMakeFiles\arrow_shared.dir\compute\exec\util.cc.obj /Fdsrc\arrow\CMakeFiles\arrow_shared.dir\ /FS -c D:\buildtrees\arrow\src\rrow-8.0.0-0091dfae41.clean\cpp\src\arrow\compute\exec\util.cc
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.31.31103\include\memory(3124): error C2027: use of undefined type 'arrow::internal::tracing::SpanImpl'
D:\buildtrees\arrow\src\rrow-8.0.0-0091dfae41.clean\cpp\src\arrow/util/tracing.h(30): note: see declaration of 'arrow::internal::tracing::SpanImpl'
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.31.31103\include\memory(3123): note: while compiling class template member function 'void std::default_delete<arrow::util::tracing::Span::Impl>::operator ()(_Ty *) noexcept const'
        with
        [
            _Ty=arrow::util::tracing::Span::Impl
        ]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.31.31103\include\memory(3233): note: see reference to function template instantiation 'void std::default_delete<arrow::util::tracing::Span::Impl>::operator ()(_Ty *) noexcept const' being compiled
        with
        [
            _Ty=arrow::util::tracing::Span::Impl
        ]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.31.31103\include\memory(3160): note: see reference to class template instantiation 'std::default_delete<arrow::util::tracing::Span::Impl>' being compiled
D:\buildtrees\arrow\src\rrow-8.0.0-0091dfae41.clean\cpp\src\arrow/util/tracing.h(64): note: see reference to class template instantiation
````

**Comment by ianmcook** [comment]
@JonLiu1993 Thank you. We are working in https://github.com/apache/arrow/pull/13108 to solve some issues caused by the upgrade to MSVC 2022. When those are resolved I will pull the changes into this PR

**Comment by ianmcook** [comment]
I tested the features locally with triplets `x64-windows` and `x64-windows-static` and observed no regressions from the current version (7.0.0).

**Comment by ianmcook** [comment]
@GPSnoopy @jgiannuzzi here's the Arrow 8.0.0 update PR. Let me know if you encounter any issues when trying this with ParquetSharp.

**Review by JonLiu1993 — APPROVED** [review]

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
I took this opportunity to minimize the patch files -- could you please pull my branch into this PR?
```
git fetch https://github.com/ras0219-msft/vcpkg dev/roschuma/arrow-8.0.0
git merge FETCH_HEAD
```

Thanks for the PR!

**Comment by ianmcook** [comment]
> I took this opportunity to minimize the patch files -- could you please pull my branch into this PR?
> 
> ```
> git fetch https://github.com/ras0219-msft/vcpkg dev/roschuma/arrow-8.0.0
> git merge FETCH_HEAD
> ```
> 
> Thanks for the PR!

Done, thanks!

**Comment by ianmcook** [comment]
> > I took this opportunity to minimize the patch files -- could you please pull my branch into this PR?
> > ```
> > git fetch https://github.com/ras0219-msft/vcpkg dev/roschuma/arrow-8.0.0
> > git merge FETCH_HEAD
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Thanks for the PR!
> 
> Done, thanks!

Pulling in those changes caused the CI to fail. I reverted them.

**Comment by ras0219-msft** [comment]
I was able to repro the issue and fix it -- could you please reset to my branch?
```
git fetch https://github.com/ras0219-msft/vcpkg dev/roschuma/arrow-8.0.0
git reset --hard FETCH_HEAD
```

**Comment by ianmcook** [comment]
> I was able to repro the issue and fix it -- could you please reset to my branch?
> 
> ```
> git fetch https://github.com/ras0219-msft/vcpkg dev/roschuma/arrow-8.0.0
> git reset --hard FETCH_HEAD
> ```

Thanks, that fixed it. I'm going to test the features locally to make sure it didn't cause any feature regressions.

**Comment by ianmcook** [comment]
I re-tested the features locally and found no regressions.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Thanks again!

File: ports/arrow/vs-2022-fixes.patch [hunk 1]
```diff
@@ -0,0 +1,42 @@
+diff --git a/cpp/src/arrow/compute/light_array_test.cc b/cpp/src/arrow/compute/light_array_test.cc
```
Comment (pitrou) [inline ports/arrow/vs-2022-fixes.patch:1]: Should these fixes be contributed back? @ianmcook

**Comment by jgiannuzzi** [comment]
> @GPSnoopy @jgiannuzzi here's the Arrow 8.0.0 update PR. Let me know if you encounter any issues when trying this with ParquetSharp.

Thanks @ianmcook! We have just released [ParquetSharp 8.0.0 Beta1](https://github.com/G-Research/ParquetSharp/releases/tag/8.0.0-beta1) built with Arrow 8.0.0. We were finally able to get rid of our `Find*.cmake` files thanks to your great work 🎉

**Pull request merged**